src/scripts.c

```diff
@@ -49,24 +50,20 @@ static void update_boot(void)
 
 static void update_triggers(bool block)
 {
-	char *restart_cmd = NULL;
 	char *block_flag = NULL;
+	char *cmd = NULL;
+	char *reexec_flag = NULL;
```

Comment:
Can we get rid of some of this string_or_die stuff?
Here have 
     char *reexec_flag = "";
on the new line 64 replace
    string_or_die(&reexec_flag, "--reexec");
with
     reexec_flag = "--reexec";
and just don't add the new line 82, the free of reexec_flag. I would remove the '} else {' on line 65 and also line 66, but you could replace line 66 with
    reexec_flag = "";
and leve this line alone.

This is smaller and faster and just as clear.

Reply:
Ah that's a good cleanup, thanks.

Reply:
ditto with block_flag